public/app/plugins/datasource/azuremonitor/azure_monitor/azure_monitor_datasource.ts

```diff
@@ -238,47 +244,117 @@ export default class AzureMonitorDatasource extends DataSourceWithBackend<
-  getMetricNamespaces(query: GetMetricNamespacesQuery, globalRegion: boolean, region?: string, custom?: boolean) {
+  async getMetricNamespaces(
+    query: GetMetricNamespacesQuery,
+    globalRegion?: boolean,
+    region?: string,
+    custom?: boolean
+  ) {
+    let resourceUri: string;
+    let subscriptionId: string | undefined;
+    let resourceGroup: string | undefined;
+
+    // Extract resourceUri or build it dynamically
+    if ('resourceUri' in query && query.resourceUri) {
+      resourceUri = query.resourceUri;
+    } else {
+      resourceUri = UrlBuilder.buildAzureMonitorGetMetricNamespacesUrl(
+        this.resourcePath,
+        this.apiVersion,
+        query,
+        this.templateSrv,
+        globalRegion,
+        region
+      );
+    }
+
+    // Extract subscription and resource group
+    const resourceUriParts = resourceUri.split('/');
+    const subscriptionIndex = resourceUriParts.indexOf('subscriptions');
+    if (subscriptionIndex !== -1 && subscriptionIndex + 1 < resourceUriParts.length) {
+      subscriptionId = resourceUriParts[subscriptionIndex + 1];
+    }
+    const resourceGroupIndex = resourceUriParts.indexOf('resourceGroups');
+    if (resourceGroupIndex !== -1 && resourceGroupIndex + 1 < resourceUriParts.length) {
+      resourceGroup = resourceUriParts[resourceGroupIndex + 1];
+    }
+
+    if (resourceGroup && subscriptionId) {
+      try {
+        const observableQuery = await this.runAzureResourceGraphQuery(subscriptionId, resourceGroup);
+        const resourcesWithMetrics = await lastValueFrom(observableQuery);
+        const allowedResourceTypes = new Set(resourceTypes.map((t) => t.toLowerCase()));
+
+        // Filter resources to include only those in resourceTypes
+        const filteredResources = resourcesWithMetrics.data[0].fields[0].values.filter((type: string) =>
+          allowedResourceTypes.has(type.toLowerCase())
+        );
+
+        return filteredResources;
+      } catch (error) {
+        console.error(`Failed to run ARG query: ${error}`);
+        return [];
+      }
+    }
+
+    // Default behavior if no resource group is provided
     const url = UrlBuilder.buildAzureMonitorGetMetricNamespacesUrl(
       this.resourcePath,
       this.apiPreviewVersion,
-      // Only use the first query, as the metric namespaces should be the same for all queries
       this.replaceSingleTemplateVariables(query),
-      globalRegion,
       this.templateSrv,
+      globalRegion,
       region
     );
+
     return this.getResource(url)
       .then((result: AzureAPIResponse<MetricNamespace>) => {
         if (custom) {
           result.value = result.value.filter((namespace) => namespace.classification === 'Custom');
         }
+
         return ResponseParser.parseResponseValues(
           result,
           'properties.metricNamespaceName',
           'properties.metricNamespaceName'
         );
       })
-      .then((result) => {
```

Comment:
I don't think this can be removed, it will break how namespaces work for the sub-resources of Azure storage accounts unless something else has changed?